File: ds-01-stack/stack-demo.js
/*


*/

let stack = [];


//-- push
stack.push(1);
stack.push(2);
stack.push(3);
console.log(stack); // [1,2,3]

//-- pop
console.log(stack.pop()); //  3
console.log(stack); // [1,2];


//- peek
var top = stack[stack.length-1];

//-- empty
console.log(stack.length == 0)

File: ds-01-stack/stack-demo.go
package main

import "fmt"

func main() {
	var v rune

	//-- create stack
	stack := make([]rune, 0)

	//-- push
	stack = append(stack, v)

	//-- peek
	v = stack[len(stack)-1] // index must be non-negative in go

	//-- pop
	stack = stack[:len(stack)-1]

	//--empty
	fmt.Println(len(stack) == 0)

}
